CMakeLists.txt

```
@@ -303,7 +299,11 @@ message(STATUS "CUDA supported archs: ${CUDA_supported_archs}")
 
 set(CUDA_TARGET_ARCHS_SORTED ${CUDA_TARGET_ARCHS})
 list(SORT CUDA_TARGET_ARCHS_SORTED)
-CUDA_find_supported_arch_values(CUDA_targeted_archs ${USED_CUDA_COMPILER} ${CUDA_TARGET_ARCHS_SORTED})
+if (NOT CMAKE_CUDA_ARCHITECTURES)
```

We have a `USE_CMAKE_CUDA_ARCHITECTURES` flag (used below) that indicates our CMAKE to set CMAKE_CUDA_ARCHITECTURES instead of manually generating them. It still will generate list of all "real" architectures and than the latest one as "virtual". Do you think that other configurations are necessary?

If so, I would add this check somewhere above:

```suggestion
if (CMAKE_CUDA_ARCHITECTURES AND USE_CMAKE_CUDA_ARCHITECTURES)
  message(FATAL_ERROR "CMAKE_CUDA_ARCHITECTURES was explicitly provided and requested to be generated based on CUDA_TARGET_ARCHS (by USE_CMAKE_CUDA_ARCHITECTURES=ON). Using both options is not allowed.")
endif()

```
to prohibit using both options at the same time.

---

> 
> 
> We have a `USE_CMAKE_CUDA_ARCHITECTURES` flag (used below) that indicates our CMAKE to set CMAKE_CUDA_ARCHITECTURES instead of manually generating them. It still will generate list of all "real" architectures and than the latest one as "virtual". Do you think that other configurations are necessary?
> 
> If so, I would add this check somewhere above:
> 
> ```
> if (CMAKE_CUDA_ARCHITECTURES AND USE_CMAKE_CUDA_ARCHITECTURES)
>   message(FATAL_ERROR "CMAKE_CUDA_ARCHITECTURES was explicitly provided and requested to be generated based on CUDA_TARGET_ARCHS (by USE_CMAKE_CUDA_ARCHITECTURES=ON). Using both options is not allowed.")
> endif()
> ```
> 
> to prohibit using both options at the same time.

I haven't noticed USE_CMAKE_CUDA_ARCHITECTURES. After re-exam  cmakelist.txt I find it is not an option, so it is useless..